Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs

```diff
@@ -1009,57 +1012,73 @@ offerQuote (driverId, merchantId, merchantOpCityId) clientId DriverOfferReq {..}
-    when (expiryTimeWithBuffer < now) $ throwError (InvalidRequest "Quote can't be responded. SearchReqForDriver is expired")
-    searchReq <- QSR.findById searchTry.requestId >>= fromMaybeM (SearchRequestNotFound searchTry.requestId.getId)
-    merchant <- CQM.findById searchReq.providerId >>= fromMaybeM (MerchantDoesNotExist searchReq.providerId.getId)
-    driver <- QPerson.findById driverId >>= fromMaybeM (PersonNotFound driverId.getId)
-    driverInfo <- QDriverInformation.findById (cast driverId) >>= fromMaybeM DriverInfoNotFound
-    transporterConfig <- SCTC.findByMerchantOpCityId merchantOpCityId (Just (DriverId (cast driverId))) >>= fromMaybeM (TransporterConfigNotFound merchantOpCityId.getId)
-    throwErrorOnRide transporterConfig.includeDriverCurrentlyOnRide driverInfo
-    when (sReqFD.response == Just Reject) (throwError QuoteAlreadyRejected)
-    driverFCMPulledList <-
-      case req.response of
-        Pulled -> do
-          QSRD.updateDriverResponse (Just Pulled) Inactive req.notificationSource sReqFD.id
-          throwError UnexpectedResponseValue
-        Accept -> do
-          whenM thereAreActiveQuotes (throwError FoundActiveQuotes)
-          pullList <-
-            case searchTry.tripCategory of
-              DTC.OneWay DTC.OneWayOnDemandDynamicOffer -> acceptDynamicOfferDriverRequest merchant searchTry searchReq driver sReqFD mbBundleVersion mbClientVersion mbConfigVersion mbDevice reqOfferedValue
-              DTC.CrossCity DTC.OneWayOnDemandDynamicOffer _ -> acceptDynamicOfferDriverRequest merchant searchTry searchReq driver sReqFD mbBundleVersion mbClientVersion mbConfigVersion mbDevice reqOfferedValue
-              DTC.InterCity DTC.OneWayOnDemandDynamicOffer _ -> acceptDynamicOfferDriverRequest merchant searchTry searchReq driver sReqFD mbBundleVersion mbClientVersion mbConfigVersion mbDevice reqOfferedValue
-              DTC.Ambulance DTC.OneWayOnDemandDynamicOffer -> acceptDynamicOfferDriverRequest merchant searchTry searchReq driver sReqFD mbBundleVersion mbClientVersion mbConfigVersion mbDevice reqOfferedValue
-              _ -> acceptStaticOfferDriverRequest (Just searchTry) driver (fromMaybe searchTry.estimateId sReqFD.estimateId) reqOfferedValue merchant clientId
-          QSRD.updateDriverResponse (Just Accept) Inactive req.notificationSource sReqFD.id
-          return pullList
-        Reject -> do
-          QSRD.updateDriverResponse (Just Reject) Inactive req.notificationSource sReqFD.id
-          pure []
-    DS.driverScoreEventHandler merchantOpCityId $ buildDriverRespondEventPayload searchTry.id driverFCMPulledList
-  pure Success
+  lockRespondQuote <- Redis.tryLockRedis (offerQuoteLockKeyWithCoolDown driverId) 15
+  lockEditDestination <- Redis.tryLockRedis (editDestinationLockKey driverId) 20
+  if lockRespondQuote && lockEditDestination
```

Comment:
Don't throw error in case of `lockRespondQuote`

Reply:
Add editDestKey in whenWithLockRedis, instead of tryLockRedis